core/ordersync/ordersync.go

```diff
@@ -144,7 +144,7 @@ type Subprotocol interface {
 	// should be sent. If nextRequest is nil, the ordersync protocol is
 	// considered finished. HandleOrderSyncResponse is the implementation for the
 	// "requester" side of the subprotocol.
-	HandleOrderSyncResponse(context.Context, *Response) (nextRequest *Request, err error)
+	HandleOrderSyncResponse(context.Context, *Response) (nextRequest *Request, filteredOrders []*zeroex.SignedOrder, err error)
```

Comment:
Looks like we don't actually need the orders themselves here, just the number. Also how do you feel about calling it `numValidOrders` instead? Not all subprotocols in the future will need to "filter" the response from the provider (in fact, ideally none of them would).

```suggestion
	HandleOrderSyncResponse(context.Context, *Response) (nextRequest *Request, numFilteredOrders int, err error)
```

Reply:
Can you also add this new requirement to the documentation for the `Subprotocol` interface?